preprocess_web/code/ravens_metadata_apps/preprocess_jobs/job_util.py

```diff
@@ -258,32 +271,62 @@ def retrieve_rows_csv(request, job, **kwargs):
             print('kwargs', kwargs)
             start_row = kwargs.get('start_row')
             num_rows = kwargs.get('number_rows')
-            if job.name.lower().endswith('.tab'):
-                print("is tab file")
-                csv_data = pd.read_csv(job.source_file.path, sep='\t', lineterminator='\r')
-                print(csv_data)
-            else:
-                csv_data = pd.read_csv(job.source_file.path)
-            max_rows = len(csv_data)
-            print("the no. of rows are ", max_rows)
 
+            # -------------------------------------------
+            # Read partial file, set lines to skip, etc
+            # -------------------------------------------
+            start_row_idx = start_row - 1  # e.g. if start_row is 1, skip nothing
+            # if start_row is 10, start on index
+            # ------------------------------------------
+            # Check the errors
+            # ------------------------------------------
             error_message = []
-            start_row_idx = start_row - 1
-
-            if start_row > max_rows:
-                err = ('The request was from %s rows but only %d'
-                       ' rows were found, so default start rows = 1'
-                       ' is set') % \
-                       (start_row, max_rows)
-                error_message.append(err)
-                start_row = 1
-            elif num_rows > max_rows:
-                err = ('The request was for %s rows but only'
-                       ' %d rows were found, so number rows'
-                       ' is set to max rows') % \
-                       (num_rows, max_rows)
-                error_message.append(err)
-                num_rows = max_rows
+            job_data = job.get_metadata()
```

Comment:
adjust this similar to the initial comments , e.g. `if job_data.success:`, etc...